### PR TITLE
feat: move override config to stwrapper

### PR DESCRIPTION
**Pull request by alisher-aituarov**
## Summary of change

Moves components override to separate component unique for each recipe

## Related issues

-   https://github.com/supertokens/supertokens-auth-react/issues/616

## Documentation changes

- https://github.com/supertokens/docs/pull/562

## Checklist for important updates

-   [x] Changelog has been updated
-   [ ] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
-   [ ] Changes to the version if needed
    -   In `package.json`
    -   In `package-lock.json`
    -   In `lib/ts/version.ts`
-   [x] Had run `npm run build-pretty`
-   [x] Had installed and ran the pre-commit hook
-   [x] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.
-   [ ] If added a new recipe interface, then make sure that the implementation of it uses NON arrow functions only (like `someFunc: function () {..}`).
-   [ ] If I added a new recipe, I also added the recipe entry point into the `size-limit` section of `package.json` with the size limit set to the current size rounded up.

## Remaining TODOs for this PR

-   [x] finish writing test

### CODE REVIEW

**Comment by alisher-aituarov** [comment]
Some test fail on this branch and on 0.28 as well

File: test/unit/componentOverrides.test.tsx [hunk 1]
```diff
@@ -112,3 +128,125 @@ describe("Theme component overrides", () => {
...
+        setMockResolvesSession({
+            userId: "mock-user-id",
+            accessTokenPayload: {},
+            invalidClaims: [],
+            doesSessionExist: false,
+            loading: false,
+        });
+    });
+
+    it.skip("Should not affect other recipes components when the same component is overridden", async () => {
```
Comment (porcellus) [inline test/unit/componentOverrides.test.tsx:193]: Why is this skipped?

Comment (porcellus) [reply test/unit/componentOverrides.test.tsx:193]: Please add another version of this test without `ThirdpartyEmailPasswordComponentsOverrideProvider` and check that the default is displayed.

Comment (porcellus) [reply test/unit/componentOverrides.test.tsx:193]: Actually, the other test-case already includes checking with two different recipe specific contexts, so you can just update this test case

**Review by porcellus — CHANGES_REQUESTED** [review]
- Please manually test the updated examples (or confirm you already did)
- I'd prefer if we threw an error if the user adds overrides for a recipe that hasn't been initialized. A meaningful error message and an early failure could help devs who try to add emailpassword overrides for thirdpartyemailpassword using the wrong context provider. (Please also add a test for this)

File: lib/ts/recipe/emailpassword/componentOverrideContext.tsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { createGenericComponentsOverrideContext } from "../../components/componentOverride/genericComponentOverrideContext";
+import { ComponentOverrideMap } from "./types";
+
+const [useOverrideContext, Provider, Consumer] = createGenericComponentsOverrideContext<ComponentOverrideMap>();
+
+export {
+    useOverrideContext,
+    Provider as RecipeComponentsOverrideContextProvider,
+    Consumer as RecipeComponentsOverrideContextConsumer,
```
[the remark this answers is not in the document]
Comment (alisher-aituarov) [reply lib/ts/recipe/emailpassword/componentOverrideContext.tsx:9]: issue is created

**Comment by alisher-aituarov** [comment]
> * Please manually test the updated examples (or confirm you already did)
> * I'd prefer if we threw an error if the user adds overrides for a recipe that hasn't been initialized. A meaningful error message and an early failure could help devs who try to add emailpassword overrides for thirdpartyemailpassword using the wrong context provider. (Please also add a test for this)

Manually tested examples that i had to update

**Comment by alisher-aituarov** [comment]
application is rendered before supertokens gets initialized, hence we are not able to access list of recipes while creating the context for recipe

**Comment by alisher-aituarov** [comment]
> @porcellus  application is rendered before supertokens gets initialized, hence we are not able to access list of recipes while creating the context for recipe. 
For now leaving as is

**Review by porcellus — APPROVED** [review]

**Review by porcellus — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [unreleased]
 
+### Changed
+
+-   Moved component override config into a recipe specific provider component
```
Comment (rishabhpoddar) [inline CHANGELOG.md:12]: add migration guide

File: CHANGELOG.md [hunk 4]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [unreleased]
 
+### Changed
```
Comment (rishabhpoddar) [inline CHANGELOG.md:10]:
```suggestion
### Breaking changes
```

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -7,6 +7,67 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [unreleased]
 
+### Breaking changes
+
+-   Moved component override config into a recipe specific provider component
+
+### Migration
+
+#### EmailVerification recipe init
```
Comment (rishabhpoddar) [inline CHANGELOG.md:16]:
```suggestion
```

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

**Review by rishabhpoddar — APPROVED** [review]

**Review by rishabhpoddar — APPROVED** [review]

**Pull request merged**